ynr/apps/candidates/models/popolo_extra.py

```diff
@@ -242,6 +242,20 @@ def has_results(self):
             return True
         return False
 
+    @property
+    def uncontested(self):
+        if not self.candidates_locked:
+            return False
+        if self.get_winner_count == self.membership_set.count():
+            return True
+        return False
+
+    def mark_uncontested_winners(self, log=True):
+        self.membership_set.update(elected=True)
```

Comment:
I had thought that this method would contain all the logic - and therefore mean you didn't need to have a conditional in the view to check if the ballot is uncontested

e.g. 
```
def mark_uncontested_winners(self, log=True):
    """
    If the election is uncontested mark all candidates as elected
    """
    if not self.uncontested:
        return
    
    self.membership_set.update(elected=True)
    
    if not log:
        return
    
    LoggedAction.objects.create(
        user=self.request.user,
        action_type="constituency-{}".format(suffix),
        ip_address=get_client_ip(self.request),
        ballot=ballot,
        source="Ballot locked; Election is uncontested",
    )
```

Then this method can be used in the `ToggleLockView` when a ballot is locked without any conditional after the suggested locks have been deleted:
```
# line 253 elections/views.py
ballot.suggestedpostlock_set.all().delete()
ballot.mark_uncontested_winners()
```

And also in the management command - where we can use the `log` argument to avoid creating LoggedAction objects:

```
def handle(self, *args):
    for ballot in Ballot.objects.all():
        ballot.mark_uncontested_winners(log=False)
```

Reply:
Ok, makes sense. I took the LoggedAction out due to the number of imports it was using, but can certainly add it back.

Reply:
If you do want to keep it in the view then its fine to do so - but probably not worth having the `mark_uncontested_winners` method on Ballot then as it isn't doing anything that you can't do directly in the view